src/@types/relay-compiler.d.ts

```diff
@@ -0,0 +1,28 @@
+import * as relayCompiler from "relay-compiler";
+
+declare module "relay-compiler" {
+  export interface Schema {
+    expectStringType: () => ScalarTypeID;
+    getEnumValues: (type: EnumTypeID) => string[];
+    getFieldName: (field: FieldID) => string;
+    getFields: (type: TypeID) => FieldID[];
+    getFieldType: (field: FieldID) => TypeID;
+    getListItemType: (type: TypeID) => TypeID;
+    getNullableType: (type: TypeID) => TypeID;
+    getTypeString: (type: TypeID) => string;
+    isAbstractType: (type: TypeID) => boolean;
```

Comment:
we can use [typescript guards][type-guards] here

e.g.
```ts
isAbstractType: (type: TypeID) => type is (UnionType | InterfaceType);
isEnum: (type: TypeID) => type is EnumType;
``` 

[type-guards]: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards